pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -536,6 +536,45 @@ func TestPriorityQueue_addToActiveQ(t *testing.T) {
 	}
 }
 
+func TestPriorityQueue_flushBackoffQCompleted(t *testing.T) {
+	tests := []struct {
+		name                  string
+		plugins               []framework.PreEnqueuePlugin
+		pod                   *v1.Pod
+		operations            []operation
+		wantUnschedulablePods int
+	}{
+		{
+			name:    "preEnqueue plugin registered, not running preEnqueue plugin when backoff completed",
+			plugins: []framework.PreEnqueuePlugin{&preEnqueuePlugin{allowlists: []string{"foo"}}},
+			pod:     st.MakePod().Name("bar").Label("bar", "").Obj(),
+			operations: []operation{
+				addPodBackoffQ,
+				flushBackoffQ,
+			},
+			wantUnschedulablePods: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			m := map[string][]framework.PreEnqueuePlugin{"": tt.plugins}
+			c := testingclock.NewFakeClock(time.Now())
+			q := NewTestQueueWithObjects(ctx, newDefaultQueueSort(), []runtime.Object{tt.pod}, WithPreEnqueuePluginMap(m),
+				WithPodInitialBackoffDuration(time.Second*1), WithPodMaxBackoffDuration(time.Second*60), WithClock(c))
+			for _, op := range tt.operations {
+				op(q, newQueuedPodInfoForLookup(tt.pod))
+			}
+			if tt.wantUnschedulablePods != len(q.unschedulablePods.podInfoMap) {
```

Comment:
To make the test robust, I'd pursue checking whether the preEnqueuPlugin was really called or not (you can add a `called int` to the test plugin). Otherwise, this test cannot detect some potential regression that calls preEnqueue but still moves the pod to activeQ/backoffQ.